File: backend/bettercollected_backend_server/app/models/workspace.py
from typing import List, Optional
import datetime as dt

from beanie import PydanticObjectId
from pydantic import BaseModel


class WorkspaceRequestDto(BaseModel):
    """Model for creating or updating a workspace."""

    title: Optional[str]
    workspaceName: Optional[str]
    description: Optional[str]
    ownerId: Optional[PydanticObjectId]
    profileImage: Optional[str]
    bannerImage: Optional[str]
    customDomain: Optional[str]


class Workspace(WorkspaceRequestDto):
    """Model for storing information about a workspace."""

    created_at: Optional[dt.datetime]
    updated_at: Optional[dt.datetime]


class WorkspaceFormSettings(BaseModel):
    """Model for storing the form settings of a workspace."""

    customUrl: Optional[str]
    responseDataOwnerField: Optional[str]
    pinned: bool = False
    roles: List[str] = []


class WorkspaceResponseDto(WorkspaceRequestDto):
    """Model for returning information about a workspace."""

    id: Optional[PydanticObjectId]


class WorkspaceFormPatch(BaseModel):
    """Model for patching the form settings of a workspace."""

    form_id: str
    pinned: Optional[bool] = False

File: webapp/src/store/jotai/workspace.ts
'use client';

import { atom, useAtom } from 'jotai';

export interface Workspace {
    id: string;
    title: string;
    workspaceName: string;
    description: string;
    ownerId: string;
    profileImage?: string;
    bannerImage?: string;
    customDomain?: string;
    dashboardAccess?: string;
    default?: string;
    disabled?: string;
    privacyPolicy?: string;
    termsOfService?: string;
    isPro?: boolean;
}

const workspaceAtom = atom<Workspace>({
    id: '',
    title: '',
    workspaceName: '',
    description: '',
    ownerId: ''
});

export default function useWorkspace() {
    const [workspace, setWorkspace] = useAtom(workspaceAtom);

    return {
        workspace,
        setWorkspace
    };
}
